### PR TITLE
Location unit test

### LINKED ISSUE #24: (6) Write test to LocationsController

### PR DESCRIPTION
**Pull request by NJohansen**
Resolve #24

### CODE REVIEW

File: TaskHouseUnitTests/FakeLocationRepository.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            new Location()
+            {
+              Id = 3,
+              Country = "Country3",
+              City = "City3",
+              ZipCode = "ZipCode3" ,
+              PrimaryLine = "PrimaryLine3" ,
+              SecondaryLine = "SecondaryLine3" ,
+              },
+            };
```
Comment (kvartborg) [inline TaskHouseUnitTests/FakeLocationRepository.cs]: Something is wrong with the indentation in this list.

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    public LocationsControllerUnitTests()
+    {
+      repo = new FakeLocationRepository();
+      controller = new LocationsController(repo);
+    }
+
+    //Test retrieve all in repository
+    [Fact]
+    public async void LocationsController_ReturnAllElementsInRepo()
```
Comment (grammeaway) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: Add CRUD-type (Get), and circumstances, so it matches the other  method names 😄

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      //Act
+      resultset = await controller.Get();
+
+      //Assert - Checks the size of the list matches with 3 which is the size
+      Assert.Equal(3,resultset.Count());
+    }
+
+    [Fact]
+    //Test GET with id
+    public async void LocationsController_ReturnObject_WhenIdIsValid()
```
Comment (grammeaway) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: Same as above, add that it's on the Get-method 🇲🇳

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    {
+      //Arrange id for location object
+      int locationId = 1;
+
+      //Fact
```
Comment (grammeaway) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: Act*

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    {
+      //Arrange
+      Location location = null;
+      int Id = 100;
+
+      //Act
+      var result = await controller.Update(Id, location);
+
+      //Assert
+      var assertResult = Assert.IsType<BadRequestResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: There is no need to assign this to a variable.

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        Country = "UpdatedCountry"
+      };
+
+      //Act
+      var result = await controller.Update(location.Id,location);
+      var resultAsObject = await controller.Get(location.Id) as ObjectResult;
+      var resultObject = resultAsObject.Value as Location;
+
+      //Assert
+      var assertResult = Assert.IsType<NoContentResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    public async void LocationController_Create_ReturnsBadRequest_WhenObjectIsNull()
+    {
+      //Arrange new ObjectResult
+      Location location = null;
+
+      //Act
+      var result = await controller.Create(location);
+
+      //Assert
+      var assertResult = Assert.IsType<BadRequestObjectResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+      var location = new Location();
+      location.Country = "Contry100";
+
+      //Act
+      var result = await controller.Create(location);
+      var resultAsObject = await controller.Create(location) as ObjectResult;
+      var resultObject = resultAsObject.Value as Location;
+
+      //Assert
+      var assertResult = Assert.IsType<ObjectResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+      //Arrange id for location object
+      int locationId = 1;
+
+      //Fact
+      var result = await controller.Get(locationId);
+      var resultAsObject = await controller.Get(locationId) as ObjectResult;
+      var resultObject = resultAsObject.Value as Location;
+
+      //Assert - Checks if the returned object is the same type and then checks id
+      var assertResult = Assert.IsType<ObjectResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+      {
+        Id = 100,
+        Country = "UpdatedCountry"
+      };
+
+      //Act
+      var result = await controller.Update(location.Id,location);
+
+      //Assert
+      var assertResult = Assert.IsType<NotFoundResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+    {
+      //Arrange
+      int Id = 1;
+
+      //Act
+      var result = await controller.Delete(Id);
+      var getDeletedResult = await controller.Get(Id);
+
+      //Assert
+      var assertResult = Assert.IsType<NoContentResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+      var assertDeletedResult = Assert.IsType<NotFoundResult>(getDeletedResult);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+    public async void LocationController_Delete_ReturnsNotFoundResult_WhenIdInvalid()
+    {
+      //Arrange
+      int Id = 100;
+
+      //Act
+      var result = await controller.Delete(Id);
+
+      //Assert
+      var assertResult = Assert.IsType<NotFoundResult>(result);
```
Comment (kvartborg) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: This should not be assigned to at var

File: TaskHouseUnitTests/LocationsControllerUnitTests.cs [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+
+    public LocationsControllerUnitTests()
+    {
+      repo = new FakeLocationRepository();
+      controller = new LocationsController(repo);
+    }
+
+    //Test retrieve all in repository
+    [Fact]
+    public async void LocationsController_Get_ReturnAllElementsInRepo()
```
Comment (grammeaway) [inline TaskHouseUnitTests/LocationsControllerUnitTests.cs]: Needs circumstances, e.g. LocationsController_Get_ReturnAllElementsInRepo_WhenGivenNoParameters()

**Pull request merged**